timer@Severga/files/timer@Severga/applet.js

```diff
@@ -12,530 +12,551 @@ const Mainloop = imports.mainloop;
+      let h = Math.floor(this.preset[i] / 3600);
+      let m = Math.floor(this.preset[i] % 3600 / 60);
+      let s = this.preset[i] % 60;
+      this.presetMenuItem[i] = new PopupMenu.PopupMenuItem((h < 10 ? "0" : "") + h + ":" + (m < 10 ? "0" : "") + m + ":" + (s < 10 ? "0" : "") + s, {});
+      this.presetMenuItem[i].connect("activate", function () {
+        this.applet._remove_timeout();
+        this.applet._remove_flashTimeout();
+        this.applet.actor.set_style("");
+        this.applet.set_applet_label("");
+        this.applet.set_applet_tooltip(_("Timer: inactive"));
+        this.applet.sliderMenuItem.setValue(0);
+
+        this.applet.period = this.preset;
+        this.applet.periodH = Math.floor(this.preset / 3600);
+        this.applet.periodM = Math.floor(this.preset % 3600 / 60);
+        this.applet.periodS = this.preset % 60;
+        this.applet.endTime = new Date(Date.now() + this.preset * 1000);
+        if (this.preset) this.applet._tick();
+        else this.applet.timeMenuItem.setLabel(_("At:  --:--:--"));
+      }.bind({applet: this, preset: this.preset[i]}));
+      // Preset item delete button
+      let del=new St.Button({label: _("[ × ]")});
+      del.connect("clicked", function () {
+        this.applet.preset.splice(this.i, 1);
+        this.applet.settings.setValue("preset", this.applet.preset);
+        this.applet._createPresetsMenuItem();
+      }.bind({applet: this, i: i}));
+      this.presetMenuItem[i].addActor(del);
+      this.presetsMenuItem.menu.addMenuItem(this.presetMenuItem[i]);
+    }
+  },
+
+  on_hplus_clicked: function() {
+    if (this.hSpin.value < 23) this.hSpin.value++;
+    else this.hSpin.value = 0;
+    this.hSpin.text = (this.hSpin.value < 10 ? "0" : "") + this.hSpin.value;
+  },
+
+  on_hminus_clicked: function() {
+    if (this.hSpin.value > 0) this.hSpin.value--;
+    else this.hSpin.value = 23;
+    this.hSpin.text = (this.hSpin.value < 10 ? "0" : "") + this.hSpin.value;
+  },
+
+  on_mplus_clicked: function() {
+    if (this.mSpin.value < 59) this.mSpin.value++;
+    else this.mSpin.value = 0;
+    this.mSpin.text = (this.mSpin.value < 10 ? "0" : "") + this.mSpin.value;
+  },
+
+  on_mminus_clicked: function() {
+    if (this.mSpin.value > 0) this.mSpin.value--;
+    else this.mSpin.value = 59;
+    this.mSpin.text = (this.mSpin.value < 10 ? "0" : "") + this.mSpin.value;
+  },
+
+  on_splus_clicked: function() {
+    if (this.sSpin.value < 59) this.sSpin.value++;
+    else this.sSpin.value = 0;
+    this.sSpin.text = (this.sSpin.value < 10 ? "0" : "") + this.sSpin.value;
+  },
+
+  on_sminus_clicked: function() {
+    if (this.sSpin.value > 0) this.sSpin.value--;
+    else this.sSpin.value = 59;
+    this.sSpin.text = (this.sSpin.value < 10 ? "0" : "") + this.sSpin.value;
+  },
+
+  on_applet_clicked: function(event) {
+    this._remove_flashTimeout();
+    this.actor.set_style("");
+    if (this.openSubmenu == "Quick setup") this.quickMenuItem.menu.open();
```

Comment:
Does this work right with localization?

Reply:
@Severga 
Yes, it works. It doesn't work if you translate it. This is the subject of this commit: https://github.com/linuxmint/cinnamon-spices-applets/pull/4688/commits/0a22e9d5d007e212d04d91a617ca0b471cdaa28f